tests/data_sources/test_geodataframe_data_source.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+
+import geopandas as gpd
+import numpy as np
+import pytest
+from pyogrio.errors import DataSourceError
+
+from hydromt.data_sources.geodataframe_data_source import GeoDataFrameDataSource
+
+
+class TestGeoDataFrame:
+    @pytest.fixture(scope="class")
+    def example_source(
+        self, geodf: gpd.GeoDataFrame, tmp_dir: Path
+    ) -> GeoDataFrameDataSource:
+        fn_gdf = str(tmp_dir / "test.geojson")
+        geodf.to_file(fn_gdf, driver="GeoJSON")
+        return GeoDataFrameDataSource(
+            name="geojsonfile",
+            driver="pyogrio",
+            metadata_resolver="convention_resolver",
+            uri=fn_gdf,
+        )
+
+    def test_read_data(
+        self, geodf: gpd.GeoDataFrame, example_source: GeoDataFrameDataSource
+    ):
```

Comment:
not really about this code, but couldn't find a better place to put this comment. I think these tests are really nice. If possible I'd also like to see some tests for the resolver so we can more a little bit more towards having unit tests as well. This is a really nice setup thought!